modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -103,11 +138,43 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
             .asScala
             .zipWithIndex
             .map { case (value, index) =>
-              EnumValue(value.getValue(), index, value.getName().asScala)
+              EnumValue(
+                value.getValue(),
+                index,
+                EnumUtil.enumValueClassName(
+                  value.getName().asScala,
+                  value.getValue,
+                  index
+                )
+              )
             }
             .toList
           Enumeration(shape.name, shape.name, values).some
-        case _ => super.stringShape(shape)
+        case _ => getDefault(shape)
+      }
+
+      override def enumShape(shape: EnumShape): Option[Decl] = {
+        val values = shape
+          .getEnumValues()
+          .asScala
+          .zipWithIndex
+          .map { case ((name, value), index) =>
+            EnumValue(value, index, name)
```

Comment:
If smithy enum values contain traits (I'm not sure it's the case), the traits should be captured here, and rendered in the Scala code